libvcs4j-spoon/src/main/java/de/unibremen/informatik/st/libvcs4j/spoon/SpoonModel.java

```diff
@@ -0,0 +1,336 @@
+package de.unibremen.informatik.st.libvcs4j.spoon;
+
+import de.unibremen.informatik.st.libvcs4j.RevisionRange;
+import de.unibremen.informatik.st.libvcs4j.Validate;
+import de.unibremen.informatik.st.libvcs4j.FileChange;
+import de.unibremen.informatik.st.libvcs4j.VCSFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spoon.Launcher;
+import spoon.SpoonModelBuilder;
+import spoon.reflect.CtModel;
+import spoon.reflect.cu.CompilationUnit;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.declaration.CtPackage;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.declaration.CtTypeParameter;
+import spoon.reflect.factory.CompilationUnitFactory;
+import spoon.reflect.factory.Factory;
+import spoon.reflect.visitor.filter.TypeFilter;
+import spoon.support.compiler.FileSystemFile;
+import spoon.support.compiler.FilteringFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.nio.file.Files;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.FileVisitResult;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.List;
+import java.util.Collection;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.lang.System.currentTimeMillis;
+
+public class SpoonModel {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SpoonModel.class);
+
+    private Optional<CtModel> model = Optional.empty();
+
+    private File temporaryDirectory = null;
+
+    private boolean noClasspath = true;
```

Comment:
Due to being a constant, I suggest to make this field static final and to write its name in capital letters (similar to `LOGGER`).